docs/using-complex.md

```diff
@@ -0,0 +1,241 @@
+and letting datalark "figure it out":
+
+[testmark]:# (skipme/kitchensink/val1/script.various/representation-shorter)
+```python
+print(mytypes.Alpha("gamma:1,2"))
+```
+
+(TODO: currently fails within union constructor: needs to support Rule 2.)
+
+We only really need the very first type to be indicated by a constructor call;
+from there, we see a positional argument (and know what type to expect for that);
+and then can apply the same logic as in the previous example (because the value is a string,
+it clearly isn't the type-level value; so, use the representation-level processing).
+
+See [The Decision Tree For Mode](consturctors.md#the-decision-tree-for-mode) docs for more info on the rules demonstrated here.
+
+
+### Kitchen Sink constructed by Mix of Levels (Explicitly)
+
+In this example, we use a bunch of explicit variations of constructor functions.
+
+This is more verbose than any of the other examples,
+but allows the author to be extremely clear about their intentions.
+
+[testmark]:# (skipme/kitchensink/val1/script.various/explicit-mixed-level)
+```python
+# First of all, let's explicitly use a representation mode constructor, for fun:
+print(mytypes.Alpha.Repr(
+	# This constructor call is going to use restructuring style.
+	# Note that the map key is "b" rather than "beta", because we're at representation level.
+	_={"b":
+		# At this moment, the "explicit mode" is still repr-level.
+		# However, using another constructor call gives us an opportunity to switch,
+		# and we'll even use an explict constructor mode to do so:
+		mytypes.Beta.Typed(_={
+			# Note the map key here is "Delta" (the type name),
+			#  not "delta:" (which would be the representation-level discriminator string).
+			"Delta": {"x": "1", "y": "2"},
+			# Note that we used the type-level representation for the Delta value;
+			# in this case we couldn't have used "1,2" and counted on Rule 2 to fix it up for us,
+			# because we're still within a context where we explicitly said we're using typed mode.
+		})
+	}
+))
+```
+
+(TODO: the syntax for explicit constructor level is a proposal and subject to review.)
+
+There's little reason to do this in this example data.
+However, you can imagine how in some circumstances, the explicitness might be important:
+for example, if a struct with a map representation strategy has type-level field names that differ
+from the keys used in the map representation (e.g. the "rename" feature is used),
+then explicitly stating you want to use one level or the other may be important.
+
+
+### Kitchen Sink constructed by Mix of Levels and Mixed Explicitness
+
+We can use explicit constructors in some positions, and eschew them in others.
+
+When we do this, the explicitly set mode continues for all the data processed
+by that constructor function (and can be reset by using another constructor).
+
+[testmark]:# (skipme/kitchensink/val1/script.various/mixed-explicitness)
+```python
```

Comment:
This code block is just a duplicate of the previous block. Copy paste mistake?

Reply:
... completely replaced.  Yeah, this was less done than I thought.  Sorry / good eyes :)

Reply:
... Oof!  I edited it, and then edited more, and evolved it back into being almost identical except for comments.

Maybe this hunk isn't needed after all.

Tell you what: if you can see a way to make this use another explicit call (probably another `.Repr` instead of `.Typed`), in a way that seems education or to probe an edge case, then go ahead and edit it to that?  Otherwise... toss it in the bin.